ProjectExample/Final_Project_Example.Rmd

```diff
@@ -133,39 +159,87 @@ ggplot(data = int_Country,
-
 ```
 
+
 # Data Analysis
 
 I then decided to perform a simple analysis to compare countries in different regions. I decided to compare countries in `Western Europe` with those of `East Asia`.
 
 I first filtered the data to create two new data sets for these regions only and then performed a t test without assuming equal variance between the scores for the countries for each region.
 
 ```{r}
-West_Eur <- Happiness %>% filter(`Regional indicator` == "Western Europe")
-NAmer <-Happiness %>% filter(`Regional indicator` == "North America and ANZ")
+West_Eur <- Happiness %>% 
+  filter(`Regional indicator` == "Western Europe")
+NAmer <-Happiness %>% 
+  filter(`Regional indicator` == "North America and ANZ")
+
+t.test(x = pull(West_Eur, Happiness_score),
+       y = pull(NAmer, Happiness_score), 
+       var.equal = FALSE)
+```
+
+The p-value was greater than 0.05, suggesting that there might not be a difference in happiness scores between Western Europe and North America. 
+
+Disclaimer - this was intended to demonstrate how one might perform such an analysis. We are not claiming that there is a difference or is not a difference in happiness scores between the countries in these two regions. 
+
+## Bonus:
+
+If I wanted to perform this analysis on multiple aspects about my data more simply, I might think about making a function like the following:
+
+```{r}
+test_reg_diff <- function(aspect){
 
-t.test(x = West_Eur$Happiness_score,
-       y = NAmer$Happiness_score, 
+t.test(x = pull(West_Eur, aspect),
+       y = pull(NAmer,aspect), 
        var.equal = FALSE)
+  }
 ```
 
-The p-value was greater than 0.05, suggesting that there might not be a difference in happiness scores between Western Europe and North America. Disclaimer- this was intended to demonstrate how one might perform such an analysis. We are not claiming that there is a difference or is not a difference in happiness scores between the countries in these two regions. 
+This allows me to just specify what aspect (or variable) I would like to compare between the two regions. I can apply it by supplying the name of the column to pull from the datasets.
+
+```{r}
+test_reg_diff(aspect = c("Social support"))
+test_reg_diff(aspect = c("Healthy life expectancy"))
+```
+
+This tests if there is a colum name that is the same as what is supplied as `aspect`. This then gets pulled from each dataset for the t test.
+
+There appears to be no significant difference according to the p-value for `Social Support` or `Healthy life expectancy`, as they are less than 0.5.
+
+Disclaimer - this was intended to demonstrate how one might perform such an analysis. We are not claiming that there is a difference or is not a difference in social support or healthy life expectancy between the countries in these two regions. 
+
+## How this might get evaluated
+According to the guidelines: 
+
+1. The student described where the data came from including how the instructor could access the data. 
```

Comment:
Maybe write this out as 1/1 or +1 point?

Reply:
great idea!